src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Setup/VisualStudioComponentSubTreeProvider.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Setup;
+
+/// <summary>
+/// Adds a sub-tree beneath the project node in Solution Explorer that shows the setup components
+/// required by the project. Enabled via the <see cref="ProjectCapability.DiagnoseVisualStudioComponents"/>
+/// project capability.
+/// </summary>
+[method: ImportingConstructor]
+[Export(ExportContractNames.ProjectTreeProviders.PhysicalViewRootGraft, typeof(IProjectTreeProvider))]
+[AppliesTo(ProjectCapability.DiagnoseVisualStudioComponents)]
+internal sealed class VisualStudioComponentSubTreeProvider(
+    IProjectThreadingService threadingService,
+    UnconfiguredProject unconfiguredProject,
+    UnconfiguredSetupComponentDataSource unconfiguredSetupComponentDataSource,
+    IUnconfiguredProjectTasksService unconfiguredProjectTasksService)
+      : ProjectTreeProviderBase(threadingService, unconfiguredProject),
+        IProjectTreeProvider
+{
+    private IDisposable? _projectCapabilitiesLink;
+
+    protected override void Initialize()
+    {
+#pragma warning disable RS0030 // Do not use banned APIs
```

Comment:
Could you explain why this is necessary?

Reply:
We ban `Initialize` so that you don't accidentally call it on other components. You should use `EnsureInitialized` instead. However, the banned API analyzer cannot be configured to exclude calls to the base class.

Most (all?) banned APIs have an explanation of why they're banned in `BannedSymbols.txt`.